datastream-server/src/main/java/com/linkedin/datastream/server/CoordinatorEventBlockingQueue.java

```diff
@@ -5,47 +5,101 @@
  */
 package com.linkedin.datastream.server;
 
+import java.util.ArrayList;
+import java.util.Arrays;
 import java.util.HashSet;
+import java.util.List;
 import java.util.Queue;
 import java.util.Set;
+import java.util.concurrent.ConcurrentHashMap;
 import java.util.concurrent.LinkedBlockingQueue;
 
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
+import com.codahale.metrics.Counter;
+import com.codahale.metrics.Gauge;
+import com.codahale.metrics.MetricRegistry;
+
+import com.linkedin.datastream.metrics.BrooklinCounterInfo;
+import com.linkedin.datastream.metrics.BrooklinGaugeInfo;
+import com.linkedin.datastream.metrics.BrooklinMetricInfo;
+import com.linkedin.datastream.metrics.DynamicMetricsManager;
+import com.linkedin.datastream.metrics.MetricsAware;
+
 
 /**
- * A blocking queue for {@link Coordinator} events
+ * A blocking queue for {@link Coordinator} events. Includes two metrics, a
+ * {@link Gauge} and {@link Counter}. The gauge provides the queue size and
+ * the counter increments when duplicate events are {@code put()}.
+ *
  * @see CoordinatorEvent.EventType
  */
-public class CoordinatorEventBlockingQueue {
+public class CoordinatorEventBlockingQueue implements MetricsAware {
 
   private static final Logger LOG = LoggerFactory.getLogger(CoordinatorEventBlockingQueue.class.getName());
+  private static final Set<BrooklinMetricInfo> METRIC_INFOS = ConcurrentHashMap.newKeySet();
+
+  static final String COUNTER_KEY = "duplicateEvents";
+  static final String GAUGE_KEY = "queuedEvents";
+
   private final Set<CoordinatorEvent> _eventSet;
   private final Queue<CoordinatorEvent> _eventQueue;
+  private final DynamicMetricsManager _dynamicMetricsManager;
+  private final Gauge<Integer> _gauge;
+  private final Counter _counter;
 
   /**
    * Construct a blocking event queue for all types of events in {@link CoordinatorEvent.EventType}
+   *
+   * @deprecated Use of this constructor should be replaced with the single parameter variant.
    */
+  @Deprecated
```

Comment:
I wouldn't bother adding this, since `CoordinatorEventBlockingQueue` is only initialized in the datastream-server module and you already integrated this change in `Coordinator`. Doesn't hurt to have it though.

Reply:
Yes, I think that makes more sense (i.e. removing the public access modifier from the queue class)